doc-content/shared-kie-docs/src/main/asciidoc/KieServer/controller-rest-api-instances-ref.adoc

```diff
@@ -0,0 +1,271 @@
+// To reuse this module, ifeval the title to be more specific as needed.
+
+[id='controller-rest-api-instances-ref_{context}']
+= {KIE_SERVER} instances and KIE containers
+
+// The {CONTROLLER} REST API supports the following endpoints for managing {KIE_SERVER} instances (remote servers) and associated KIE containers. The {CONTROLLER} REST API base URL is `\http://SERVER:PORT/CONTROLLER/rest/controller/`. All requests require basic HTTP Authentication or token-based authentication for the `rest-all` user role if you installed {CENTRAL} and you want to use the built-in {CONTROLLER}, or the `kie-server` user role if you installed the {HEADLESS_CONTROLLER} separately from {CENTRAL}.
+
+== [GET] /runtime/servers/{serverId}/instances
+
+Returns all {KIE_SERVER} instances configured with the {CONTROLLER} for a specified {KIE_SERVER} template.
+
+.Request parameters
+[cols="25%,45%,15%,15%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`serverId`
+|ID of the {KIE_SERVER} template for which you are retrieving {KIE_SERVER} instances
+|String
+|Required
+|===
+
+.Example server response (JSON)
+[source,json]
+----
+{
+  "server-instance-key": [
+    {
+      "server-instance-id": "default-kieserver-instance@localhost:8080",
+      "server-name": "default-kieserver-instance@localhost:8080",
+      "server-template-id": "default-kieserver",
+      "server-url": "http://localhost:8080/kie-server/services/rest/server"
+    }
+  ]
+}
+----
+
+== [GET] /runtime/servers/{serverId}/containers/{containerId}/instances
+
+Returns all KIE containers for a specified {KIE_SERVER} template in the {KIE_SERVER} instance.
```

Comment:
This method is to return all container instances with giving containerId, in all Kie Server instances.

Reply:
Something like:
Returns all KIE container instances for a specified KIE container in the KIE Server template

Reply:
I've updated to "Returns all instances of a specified KIE container in a specified {KIE_SERVER} template."